dataplane/base.h

```diff
@@ -14,13 +14,48 @@
 namespace dataplane::base
 {
 
+class PortMapper
+{
+	static constexpr tPortId INVALID_PORT_ID = std::numeric_limits<tPortId>::max();
+	uint16_t ports_count_ = 0;
+	tPortId dpdk_ports_[CONFIG_YADECAP_PORTS_SIZE] = {INVALID_PORT_ID}; // logical to dpdk
+	tPortId logical_ports_[std::numeric_limits<tPortId>::max() + 1] = {INVALID_PORT_ID}; // dpdk to logical
+public:
+	uint16_t size() const { return ports_count_; }
+	std::optional<tPortId> Register(tPortId dpdk_port)
+	{
+		if (ports_count_ < CONFIG_YADECAP_PORTS_SIZE)
+		{
+			if (logical_ports_[dpdk_port] == INVALID_PORT_ID)
+			{
+				logical_ports_[dpdk_port] = ports_count_;
+				dpdk_ports_[ports_count_] = dpdk_port;
+				return std::optional<tPortId>{ports_count_++};
+			}
+			else
+			{
+				YANET_LOG_ERROR("Duplicate dpdk port id provided to PortMapper");
+				return {};
+			}
+		}
+		else
+		{
+			YANET_LOG_ERROR("CONFIG_YADECAP_PORTS_SIZE exceeded");
+			return {};
+		}
+	}
```

Comment:
IMHO nesting provides better clarity. Maybe we should discuss a threshold?

Reply:
It's generally easier to follow the code when we handle edge cases first and get them out of the way.
This way, the "happy path" stands out without being buried in nested ifs.
It's like "If anything's wrong, let's deal with it right away; otherwise, here's what we're really here to do."

But of course it's not that important, so you can leave it as it is if you want.

However, choosing max column width is a great idea (we will choose it when we adopt some coding standard, of course)

Reply:
Let's discuss this later, while choosing codestyle